File: src/main/kotlin/com/smody/book/book/service/BookService.java
package com.smody.book.book.service;

import com.smody.book.book.api.BookApi;
import com.smody.book.book.api.BookApiResponse;
import com.smody.book.book.dto.BookResponse;
import java.util.List;
import java.util.stream.Collectors;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
public class BookService {

    private final BookApi bookApi;

    public List<BookResponse> findAllByTitle(String title) {
        return convertToResponse(bookApi.findAllByTitle(title));
    }

    private List<BookResponse> convertToResponse(List<BookApiResponse> apiResponse) {
        return apiResponse.stream()
                .map(item -> new BookResponse(item.title(), item.image(), item.author(),
                        item.pubdate(),
                        item.publisher(), item.description()))
                .collect(Collectors.toList());
    }
}

File: src/main/kotlin/com/smody/book/book/service/BookService.kt
package com.smody.book.book.service

import com.smody.book.book.api.BookApi
import com.smody.book.book.api.BookApiResponse
import com.smody.book.book.dto.BookResponse
import org.springframework.beans.factory.annotation.Qualifier
import org.springframework.web.bind.annotation.RestController

@RestController
class BookService(
    @Qualifier("naver") private val bookApi: BookApi
) {
    fun findAllByTitle(title: String): List<BookResponse> {
        return convertToResponse(bookApi.findAllByTitle(title))
    }

    private fun convertToResponse(apiResponse: List<BookApiResponse>): List<BookResponse> {
        return apiResponse.map { item ->
            BookResponse(
                item.title(), item.image(), item.author(),
                item.pubdate(),
                item.publisher(), item.description()
            )
        }
    }
}
